cni/pkg/nodeagent/server.go

```diff
@@ -69,13 +69,19 @@ func NewServer(ctx context.Context, ready *atomic.Value, pluginSocket string, ar
 		return nil, fmt.Errorf("error initializing kube client: %w", err)
 	}
 
-	cfg := &iptables.Config{
+	hostCfg := &iptables.Config{
 		RedirectDNS: args.DNSCapture,
 		EnableIPv6:  args.EnableIPv6,
 	}
 
+	podCfg := &iptables.Config{
+		RedirectDNS: args.DNSCapture,
+		EnableIPv6:  args.EnableIPv6,
+		Reconcile:   args.ReconcilePodRulesOnStartup,
```

Comment:
I am a bit confused. This is called 'ReconcilePodRulesOnStartup' but it seems to make it so we always reconcile, not just 'on startup'. 

Is it implicitly only 'on startup' since we don't even trigger the function if except on startup?

Reply:
> Is it implicitly only 'on startup' since we don't even trigger the function if except on startup?

That's what this PR does: https://github.com/istio/istio/pull/53906

You either get no reconciliation, or you get reconciliation on startup (which in ambient is the only time we would apply rules to an already-running pod anyway)

Reply:
> Is it implicitly only 'on startup' since we don't even trigger the function if except on startup?

Yes, `ReconcilePodRulesOnStartup `will set the reconciliation to the pod's iptablesConfigurator, which is used whenever we need to create in-pod rules. If a pod is enrolled in the mesh, then although reconciliation is potentially enabled, although it will not be used because no residue will be found.
The only time it is expected to be triggered is in #53906 commit when `CreateInPodRules()` is invoked on already running enrolled workloads at startup of cni-agent.

There is a scenario, however, where reconciliation could be triggered during the first enrollment of pods: if the application itself has already modified the iptables inside the pod's namespace (residues are found). In that case, the reconciliation will be triggered, it will attempt a cleanup, but nothing will be removed unless the application has created unknown ISTIO_* chains/jumps. After that, the rules will be applied.

If we plan to enforce reconciliation only for the pods at startup, then the approach will need to change slightly. I could update the code to include a new field in `CreateInPodRules(...existing args..., overrideReconciliation bool)`. This field could be used in https://github.com/istio/istio/pull/53906 if `ReconcilePodRulesOnStartup `is also set. WDYT?

Reply:
> There is a scenario, however, where reconciliation could be triggered during the first enrollment of pods: if the application itself has already modified the iptables inside the pod's namespace (residues are found). In that case, the reconciliation will be triggered, it will attempt a cleanup, but nothing will be removed unless the application has created unknown ISTIO_* chains/jumps. After that, the rules will be applied.

I think this is fine/I don't want to add additional knobs for this. Any rules with our name on them that exist before we see the pod would be considered an unrecoverable error condition today _anyway_, without this fix (and we have never seen this happen).